nap/docker/ckan/ckan-plugins/ckanext-napote_theme/ckanext/napote_theme/templates/user/snippets/login_form.html

```diff
@@ -0,0 +1,31 @@
+{#
+Renders the login form.
+
+action        - The url that the form should be submitted to.
+error_summary - A tuple/list of form errors.
+
+Example:
+
+  {% snippet "user/snippets/login_form.html", action=c.login_handler, error_summary=error_summary %}
+
+#}
+{% import 'macros/form.html' as form %}
+
+{% set username_error = true if error_summary %}
+{% set password_error = true if error_summary %}
+
+<form action="{{ action }}" method="post" class="form-horizontal">
+    {{ form.errors(errors=error_summary) }}
+
+    {{ form.input('login', label=_("Username or email"), id='field-login', value="", error=username_error, classes=["control-medium"]) }}
```

Comment:
Not a biggie, but: In the implementation logic you first check if user gave email and if email can't be found username is used. But here the help text indicates that username is used first.

Reply:
The order of the check has no affect how user logs in. You can use username or email.